aws/eks-anywhere
# Cloudstack reconciler validator

*Issue #, if available:*
- [Cloudstack validator#1055](https://github.com/aws/eks-anywhere-internal/issues/1055)
- [Validate Datacenter Config#1259](https://github.com/aws/eks-anywhere-internal/issues/1259)

*Description of changes:*
Building validator in controller to use cmk to run validations against Cloudstack
https://github.com/aws/eks-anywhere/pull/5452

*Testing (if applicable):*
manually tested: 
- manually add new secret to create new wl cluster
- successfully validate dc
- use invalid secret (invalid api-key) to create new wl cluster
  - "Reconciler error" err=<  getting domain info - 🙈 Error: failed to decode response
- use unexistent secret name to create cluster
  - cloudstack_datacenter_controller.go:73] "Failed to reconcile CloudStackDatacenterConfig" err="Secret \"secret2\" not found" controller="cloudstackdatacenterconfig"
  - Reconciler error" err="Secret \"secret2\" not found"
- fail to validate dc stops reconciliation
  - CloudStackDatacenterConfig hasn't been validated yet"


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
